backend/globaleaks/handlers/admin/user.py

```diff
@@ -37,9 +41,52 @@ def db_set_user_password(session, tid, user, password):
             if crypto_escrow_pub_key_tenant_n:
                 user.crypto_escrow_bkp2_key = Base64Encoder.encode(GCE.asymmetric_encrypt(crypto_escrow_pub_key_tenant_n, cc))
 
+def generate_analyst_key_pair(session, user, user_session):
+    if user.role not in [EnumUserRole.analyst.name, EnumUserRole.admin.name]:
+        return
+    try:
+        global_stat_prv_key, global_stat_pub_key = GCE.generate_keypair()
+
+        global_stat_pub_key_config = session.query(models.Config).filter(
+            models.Config.tid == user.tid,
+            models.Config.var_name == 'global_stat_pub_key'
+        ).first()
+
+        if not global_stat_pub_key_config:
+            raise ValueError("Config not found")
+
+        if not global_stat_pub_key_config.value:
+            global_stat_pub_key_config.value = global_stat_pub_key
+        else:
+            user_admin = session.query(models.User).filter(
+                models.User.id == user_session.attrs.get('user_id'),
+                models.User.tid == user.tid
+            ).one_or_none()
+
+            if not user_admin:
+                raise ValueError("Admin user for decryption not found")
+
+            sts_prv_key = base64.b64decode(user_admin.crypto_global_stat_prv_key)
+            global_stat_prv_key = GCE.asymmetric_decrypt(user_session.cc, sts_prv_key)
+        encrypted_global_stat_prv_key = GCE.asymmetric_encrypt(
+            user.crypto_pub_key, global_stat_prv_key
+        )
+        crypto_stat_key_encoded = Base64Encoder.encode(encrypted_global_stat_prv_key).decode()
+        session.query(models.User).filter(
+            models.User.id == user.id
+        ).update(
+            {'crypto_global_stat_prv_key': crypto_stat_key_encoded}
+        )
+    except ValueError as e:
+        logging.error(e)
+        raise errors.ForbiddenOperation
+    except Exception as e:
+        logging.error(e)
+        session.rollback()
+        raise errors.InternalServerError
 
 
-def db_create_user(session, tid, user_session, request, language):
+def db_create_user(session, tid, user_session, request, language, wizard: bool = False):
```

Comment:
todo: refactor as it breaks Separation of Concerns (SoC) or Encapsulation. Bottom layers should not be aware of upper layers.